mcs/class/System/System.Net/HttpWebRequest.cs

```diff
@@ -899,6 +899,11 @@ public override Stream EndGetRequestStream (IAsyncResult asyncResult)
 
 			try {
 				return TaskToApm.End<Stream> (asyncResult);
+			} catch (System.Threading.Tasks.TaskCanceledException tce) {
```

Comment:
We are using `OperationCanceledException` in most of the other code - if you have a specific reason to use `TaskCanceledException` instead, then please remove the ugly namespace (add `using System.Threading.Tasks` if necessary).